docs/security-features/privilege-restriction/snap-confinement.rst

```diff
@@ -0,0 +1,96 @@
+
+* AppArmor
+* Seccomp
+* Device cgroups
+* Traditional permissions and Mount namespaces
+
+See `Confinement and isolation mechanisms <https://snapcraft.io/docs/security-policies#p-2741-confinement-and-isolation-mechanisms>`_ for more details.
+
+Confinement Types 
+#################
+
+Strict
+------
+
+Strict confinement is declared in a snapcraft.yaml through ``confinement: strict``.
+This level of confinement utilises all of the confinement mechanisms listed above to ensure the snap executables and data are sandboxed, with limited access.
+
+Interfaces
+^^^^^^^^^^
+
+In order to gain access to certain resources on the host system, snap interfaces are used.
+Interfaces enable resources from one snap to be shared with another and with the system.
+Interfaces are composed by:
+
+* slots: used by snaps or the system (implicit slot) to expose resources
+* plugs: used by snaps to consume access resources exposed by a slot
+
+See `Interfaces <https://snapcraft.io/docs/interfaces>`_ for more details.
+
+Classic
+-------
+
+Classic confinement is declared in a snapcraft.yaml through ``confinement: classic``.
+This level of confinement allows access to the system’s resources in much the same way traditional packages do. Because of this, manual review is required to publish this type of snap.
+It is important to note that ``classic`` snaps **do not** run on Ubuntu core systems.
+
+Devmode
+-------
+
+Devmode confinement is declared in a snapcraft.yaml through ``confinement: devmode``.
+This level of confinement is used during the development of snaps. 
+Devmode runs snaps similarly to strictly confined snaps, however, instead of limiting access to the host system resources, it produces a debug output to enable developers to identify what access or interfaces may be needed for the snap to run as strictly confined.
+
+Confinement Mechanisms
+######################
+
+AppArmor
+--------
+
+AppArmor profiles are generated by snapd for each command in a snap and are used to restrict or allow certain capabilities for each command.
+Declaring interfaces in the snap allows the default AppArmor profile to be extended.
+`This default profile <https://github.com/canonical/snapd/blob/master/interfaces/apparmor/template.go#L63_>`_ defines a common set of rules that are applied to all bases by default.
+For example, when a snap plugs the camera interface, `this profile <https://github.com/canonical/snapd/blob/master/interfaces/builtin/camera.go#L32>`_ is added to extend the default policy.
+
+See :doc:`apparmor` for more details.
+
+Seccomp
+-------
+
+Similarly to how AppArmor is used for snaps, Seccomp filters are also generated for each command in a snap.
+These allow for processes inside the snap to have syscall filtering. Again, these can be extended through the use of snap interfaces.
+
+See :any:`seccomp-filtering` for more details.
+
+Device cgroups
+--------------
+
+Cgroups are used in snaps to resource limit processes running inside the snap.
```

Comment:
resources are usually understood as CPU, RAM, IO. This should likely state that the sandbox is using device controller only.

There is such a thing called quotas in snapd, but it's not part of the sandbox setup, but rather just a think we configure in systemd and only applies to services (apps with `daemon: ..`).